### PR TITLE
add Pattern and Matcher actual implementation for js and native

### PR DESCRIPTION
**Pull request by eymar**
Update FontFeature.kt and FontVariation.kt parse functions to use regex numeric groups instead of named groups because [k/n doesn't support named groups](https://youtrack.jetbrains.com/issue/KT-41890)

### CODE REVIEW

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/Expects.kt [hunk 1]
```diff
@@ -14,7 +14,9 @@ expect class Pattern {
 }
 
 expect class Matcher {
+    @Deprecated("Not supported in k/n. Consider using group(ix: Int)")
```
Comment (olonho) [inline skiko/src/commonMain/kotlin/org/jetbrains/skia/Expects.kt:17]: Let’s just remove it, but keep the comment.

File: skiko/src/jsMain/kotlin/org/jetbrains/skia/Actuals.js.kt [hunk 2]
```diff
@@ -8,18 +8,31 @@ actual fun <R> commonSynchronized(lock: Any, block: () -> R) {
 
 actual fun String.intCodePoints(): IntArray = TODO()
 
-actual class Pattern {
-    actual fun split(input: CharSequence): Array<String?>? = TODO()
-    actual fun matcher(input: CharSequence): Matcher = TODO()
+actual class Pattern constructor(regex: String) {
+    private val _regex = Regex(regex)
+
+    actual fun split(input: CharSequence): Array<String?>? = _regex.split(input).toTypedArray()
```
Comment (VeselovAlex) [inline skiko/src/jsMain/kotlin/org/jetbrains/skia/Actuals.js.kt:14]: Can't we return `Array<String>`, without nulls?

**Review by VeselovAlex — APPROVED** [review]

File: skiko/src/jsMain/kotlin/org/jetbrains/skia/Actuals.js.kt (hunk 2 above)
Comment (olonho) [reply skiko/src/jsMain/kotlin/org/jetbrains/skia/Actuals.js.kt:14]: Seems to be better indeed.

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/Expects.kt (hunk 1 above)
Comment (eymar) [reply skiko/src/commonMain/kotlin/org/jetbrains/skia/Expects.kt]: fixed

File: skiko/src/commonMain/kotlin/org/jetbrains/skia/FontFeature.kt [hunk 3]
```diff
@@ -78,16 +78,20 @@ class FontFeature(val _tag: Int, val value: Int, val start: UInt, val end: UInt)
         val EMPTY = arrayOfNulls<FontFeature>(0)
         val _splitPattern = compilePattern("\\s+")
         val _featurePattern =
-            compilePattern("(?<sign>[-+])?(?<tag>[a-z0-9]{4})(?:\\[(?<start>\\d+)?:(?<end>\\d+)?\\])?(?:=(?<value>\\d+))?")
+            compilePattern("([-+])?([a-z0-9]{4})(?:\\[(\\d+)?:(\\d+)?\\])?(?:=(\\d+))?")
+
+        private val groupsIx = mapOf(
```
[the remark this answers is not in the document]
Comment (eymar) [reply skiko/src/commonMain/kotlin/org/jetbrains/skia/FontFeature.kt]: fixed as you suggested.

File: skiko/src/jsMain/kotlin/org/jetbrains/skia/Actuals.js.kt (hunk 2 above)
Comment (eymar) [reply skiko/src/jsMain/kotlin/org/jetbrains/skia/Actuals.js.kt]: agree. fixed

**Review by olonho — APPROVED** [review]

**Pull request merged**